docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,312 @@
+permalink: docs/jsx-in-depth.html
+---
+
+Fundamentally, JSX just provides syntactic sugar for the `React.createElement(component, props, children)` function. The JSX code:
+
+```js
+<MyComponent prop1={'one'} prop2={'two'}>stuff</MyComponent>
+```
+
+compiles into:
+
+```js
+React.createElement(MyComponent, {prop1: 'one', prop2: 'two'}, 'stuff')
+```
+
+You can also use the self-closing form of the tag if there are no children. So:
+
+```js
+<div className={'blorp'} />
+```
+
+compiles into:
+
+```js
+React.createElement('div', {className: 'blorp'}, null)
+```
+
+If you want to test out how some specific JSX is converted into JavaScript, you can try out [the online Babel compiler](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-0&code=function%20hello()%20%7B%0A%20%20return%20%3Cdiv%3EHello%20world!%3C%2Fdiv%3E%3B%0A%7D).
+
+## Specifying The React Element Type
+
+The first part of a JSX tag determines the type of the React element.
+
+Capitalized types indicate that the JSX tag is referring to a React component. These tags get compiled into a direct reference to the named variable, so if you use the JSX `<Foo />` expression, `Foo` must be in scope.
+
+Since JSX compiles into calls to `React.createElement`, the `React` library must also always be in scope from your JSX code.
+
+For example, both of the imports are necessary in this code, even though 'React' and 'MyComponent' are not directly referenced from JavaScript:
+
+```js
+import React from 'react';
+import MyComponent from './MyComponent';
+
+export default function MyRedComponent() {
+  return <MyComponent color="red" />;
+}
+```
+
+If you don't use a JavaScript bundler and added React as a script tag, it is already in scope as a React global.
+
+You can also refer to a React component using dot-notation from within JSX. This is convenient if you have a single module that exports many React components. For example, if `MyComponents.DatePicker` is a component, you can invoke it directly from JSX with:
+
+```js
+import React from 'react';
+
+var MyComponents = {
+  DatePicker: function(props) {
+    return <div>imagine a {props.color} datepicker here</div>;
+  }
+}
+
+export default function BlueDatePicker() {
+  return <MyComponents.DatePicker color={'blue'} />;
+}
+```
+
+If the element type starts with a lowercase letter, React will try to create a React element that maps directly to an HTML tag, rather than creating an element based on a custom component. In general, name your components starting with capital letters, and this will be fine. If you do have a component that starts with a lowercase letter, assign it to a capitalized variable before using it in JSX.
+
+For example, this code will not run as expected:
+
+```js
+import React from 'react';
+
+function hello(props) {
+  // This use of <div> is legitimate because div is a valid HTML tag
+  return <div>Hello {props.toWhat}</div>;
+}
+
+export default function HelloWorld() {
```

Comment:
I think we shouldn't emphasize exports in examples unless we specifically mention module boundaries. It's fine to leave this as a function and presume that it gets rendered by something later.